src/NUnitCommon/nunit.common/ResultSummary.cs

```diff
@@ -204,10 +216,11 @@ private void Summarize(XmlNode node, bool failedInFixtureTearDown)
                         InvalidAssemblies++;
                         UnexpectedError = true;
                     }
-                    if ((type == "SetUpFixture" || type == "TestFixture") && status == "Failed" && label == "Error" && site == "TearDown")
-                    {
-                        failedInFixtureTearDown = true;
-                    }
+                    // TODO: This seems wrong and causes failures. Leaving it commented while awaiting feedback.
+                    //if ((type == "SetUpFixture" || type == "TestFixture") && status == "Failed" && label == "Error" && site == "TearDown")
+                    //{
+                    //    failedInFixtureTearDown = true;
+                    //}
```

Comment:
This causes code to increment `ErrorCount` instead of `PassCount`.
You changed the `MockAssembly.Passed` expected count by adding: `PassedButFailedInTearDown`
Should a test be marked Error if the teardown fails?

Reply:
I asked @mikkelbu for some background on that change, made seven years ago, as this was the part of the change that gave me the most trouble

ResultReporter was already using PassedInAttribute rather than Passed. I changed how the number is calculated so that Passed could be used but the values are still the same, and all tests are still passing.

@mikkelbu Can you take a look at the last commit, where I made that change?

Reply:
Your update to ResultReportedTest now expects: Failed_Errors to be 1, previously it was 7. The 6 difference is the `PassedButFailedInTearDown` you added to `Passed`

The `PassedInAttribute` was used in other tests: NUnitFrameworkDriverTests and TestAgentRunnerTests.
If that is what comes back from the engine/framework then indeed the added code in ResultSummary is wrong.
If we want tests to be considered failed if the teardown fails, then this counting might need updating in the framework.

If we keep the counting we have now, you can remove the `bool failedInFixtureTearDown` parameter.

Reply:
> I asked @mikkelbu for some background on that change, made seven years ago, as this was the part of the change that gave me the most trouble

I'm sorry, but I cannot remember anything about this. I think I only had made a handful of PRs in the console before doing this change, so I think most of the logic was somewhat new for me.

> @mikkelbu Can you take a look at the last commit, where I made that change?

I can check it tomorrow night - so in 20 hours or so - and can also take a closer look at my old PR and see if I remember some more.